### PR TITLE
[RISC-V] Fixing createdump for RISC-V

### PR DESCRIPTION
**Pull request by clamp03**
Initial patch for createdump. Fixed errors when to make dumps in RISC-V only.
I didn't check generated dump is correct yet. I will check later when others are prepared.

Part of https://github.com/dotnet/runtime/issues/84834
cc @wscho77 @HJLeee @JongHeonChoi @t-mustafin @gbalykov @tomeksowi @Sirntar @yurai007

### CODE REVIEW

**Comment by gbalykov** [comment]
cc @viewizard

**Comment by clamp03** [comment]
@jkotas Could you take a look at this PR? Thank you.

File: src/coreclr/debug/daccess/datatargetadapter.cpp [hunk 1]
```diff
@@ -131,6 +131,14 @@ DataTargetAdapter::GetPlatform(
...
         return E_NOTIMPL;
 
+#ifndef HOST_WINDOWS
+    // In windows, IMAGE_FILE_MACHINE_RISCV64 is not declared yet.
+    case IMAGE_FILE_MACHINE_RISCV64:
+        ulExpectedPointerSize = 8;
+        platform = CORDB_PLATFORM_POSIX_RISCV64;
+    break;
+#endif // HOST_WINDOWS
+
```
Comment (jkotas) [inline src/coreclr/debug/daccess/datatargetadapter.cpp:141]:
```suggestion
    case IMAGE_FILE_MACHINE_RISCV64:
        ulExpectedPointerSize = 8;
        platform = CORDB_PLATFORM_POSIX_RISCV64;
    break;

```
And instead add the following somewhere (src/coreclr/inc/clrint.h may be a good place):
```
#ifndef IMAGE_FILE_MACHINE_RISCV64
#define IMAGE_FILE_MACHINE_RISCV64           0x5064  // RISCV64
#endif
```

Comment (clamp03) [reply src/coreclr/debug/daccess/datatargetadapter.cpp]: Thank you!

**Review by jkotas — APPROVED** [review]

**Comment by jkotas** [comment]
Thank you!

**Review by gbalykov — COMMENTED** [review]
LGTM

**Pull request merged**